src/plugins/discord/update_discord_api_snapshots.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -eu
+
+snapshots_dir=src/plugins/discord/snapshots
+test_instance_url="https://discordapp.com/api"
+
+if [ ! "$(jq --version)" ]; then
+  printf >&2 'This script depends on jq. Please install it.\n'
+  exit 1
+fi
+
+if [ -z "${SOURCECRED_DISCORD_BOT_TOKEN:-}" ]; then
+  printf >&2 "Please set the SOURCECRED_DISCORD_TOKEN environment variable.\n"
+  exit 1
+fi
+
+toplevel="$(git -C "$(dirname "$0")" rev-parse --show-toplevel)"
+cd "${toplevel}"
+
+fetch() {
+  url="${test_instance_url}$2"
+  path="${snapshots_dir}/$1"
+  curl -sfL "$url" \
+    -H "Accept: application/json" \
+    -H "Authorization: Bot ${SOURCECRED_DISCORD_BOT_TOKEN}" \
+    | jq '.' > "${path}"
+}
+
+rm -r "${snapshots_dir}"
+mkdir "${snapshots_dir}"
+fetch "guilds" "/guilds/678348980639498428"
+fetch "channels" "/guilds/678348980639498428/channels"
+fetch "members" "/guilds/678348980639498428/members?after=0&limit=5"
+fetch "messages" "/channels/678348980849213472/messages?after=0&limit=5"
+fetch "messages_pagination_0" "/channels/678394406507905129/messages?after=0&limit=5"
+fetch "messages_pagination_1" "/channels/678394406507905129/messages?after=678394436757094410&limit=5"
+fetch "messages_pagination_2" "/channels/678394406507905129/messages?after=678394455849566208&limit=5"
+fetch "reaction" "/channels/678348980849213472/messages/678697350385893404/reactions/sourcecred:678399364418502669?after=0&limit=5"
```

Comment:
I notice you're manually giving the snapshots names. In the Discourse counterpart of this we encode the URL into the filename, so making the same request from the MockFetcher would find this same file.

Since this is a different approach, I wonder how you'd want something like MockFetcher to resolve these files?

Reply:
Updated